packages/chain-adapters/src/ethereum/EthereumChainAdapter.ts

```diff
@@ -191,9 +191,11 @@ export class EthereumChainAdapter implements ChainAdapter {
     return ethAddress as string
   }
 
-  async validateAddress(address: string): Promise<ValidAddressResult> {
+  validateAddress(address: string): ValidAddressResult {
     const isValidAddress = WAValidator.validate(address, this.getType())
-    if (isValidAddress) return { valid: true, result: ValidAddressResultType.Valid }
+    const isValidEnsAddress = new RegExp(/([A-Za-z]*\.)?([A-Za-z]{3,})\.eth/).test(address)
```

Comment:
Regex check might not be optimal here. As per the [EIP-137](https://eips.ethereum.org/EIPS/eip-137) spec, names can be more complex than this format, especially with internationalization. The spec recommends the [idna-uts46](https://www.npmjs.com/package/idna-uts46) lib, but [ens-normalize](`https://github.com/adraffy/ens-normalize.js`) seems to be a better fit API for these needs.

Reply:
Since reverse lookup is done in https://github.com/shapeshift/web/pull/704/files it's probably enough to test it against regex. Support for Rinkeby's `.test` domains could be added when support for testnets is added.